boa/src/syntax/lexer/cursor.rs

```diff
@@ -15,25 +15,24 @@ impl<R> Cursor<R> {
     pub(super) fn pos(&self) -> Position {
         self.pos
     }
-
-    /// Advances the position to the next line.
+    /// Advances the position to the next column.
     #[inline]
-    fn next_column(&mut self) {
+    pub(super) fn next_column(&mut self) {
         let current_line = self.pos.line_number();
         let next_column = self.pos.column_number() + 1;
         self.pos = Position::new(current_line, next_column);
     }
 
-    /// Advances the position to the next column.
+    /// Advances the position to the next line.
     #[inline]
-    fn next_line(&mut self) {
+    pub(super) fn next_line(&mut self) {
```

Comment:
I don't think this should be called from `super`. This should only be called from the cursor, when finding a "next line" character

Reply:
Yeah makes sense, when I moved the code from the old lexer over (which did make these calls) I made adjustments so that it would work while I got other parts working.

Reply:
OK, For the future, we should restrict it, I think that way we can abstract all that information from the lexer itself.

Reply:
Fixed